plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeConnectorTest.java

```diff
@@ -944,6 +945,336 @@ public void testSettingChangeDataFeedEnabledProperty()
+                "VALUES (1, 7, '{\"f1\":2,\"f2\":3,\"row_t\":{\"f1\":4,\"f2\":5,\"row_t\":{\"f1\":6,\"f2\":7}}}'), " +
+                        "(11, 17, '{\"f1\":12,\"f2\":13,\"row_t\":{\"f1\":14,\"f2\":15,\"row_t\":{\"f1\":16,\"f2\":17}}}'), " +
+                        "(21, 27, '{\"f1\":22,\"f2\":23,\"row_t\":{\"f1\":24,\"f2\":25,\"row_t\":{\"f1\":26,\"f2\":27}}}')");
+
+        // Test predicates on immediate child column and deeper nested column
+        assertQuery("SELECT id, CAST(row_t.row_t.row_t AS JSON) FROM " + tableName + " WHERE row_t.row_t.row_t.f2 = 27", "VALUES (21, '{\"f1\":26,\"f2\":27}')");
+        assertQuery("SELECT id, CAST(row_t.row_t.row_t AS JSON) FROM " + tableName + " WHERE row_t.row_t.row_t.f2 > 20", "VALUES (21, '{\"f1\":26,\"f2\":27}')");
+        assertQuery("SELECT id, CAST(row_t AS JSON) FROM " + tableName + " WHERE row_t.row_t.row_t.f2 = 27",
+                "VALUES (21, '{\"f1\":22,\"f2\":23,\"row_t\":{\"f1\":24,\"f2\":25,\"row_t\":{\"f1\":26,\"f2\":27}}}')");
+        assertQuery("SELECT id, CAST(row_t AS JSON) FROM " + tableName + " WHERE row_t.row_t.row_t.f2 > 20",
+                "VALUES (21, '{\"f1\":22,\"f2\":23,\"row_t\":{\"f1\":24,\"f2\":25,\"row_t\":{\"f1\":26,\"f2\":27}}}')");
+
+        // Test predicates on parent columns
+        assertQuery("SELECT id, row_t.row_t.row_t.f1 FROM " + tableName + " WHERE row_t.row_t.row_t = ROW(16, 17)", "VALUES (11, 16)");
+        assertQuery("SELECT id, row_t.row_t.row_t.f1 FROM " + tableName + " WHERE row_t = ROW(22, 23, ROW(24, 25, ROW(26, 27)))", "VALUES (21, 26)");
+
+        assertUpdate("DROP TABLE IF EXISTS " + tableName);
+    }
+
+    @Test
+    public void testProjectionWithCaseSensitiveField()
+    {
+        // TODO consider moving this in BaseConnectorTest
+        String tableName = "test_projection_with_case_sensitive_field_" + randomNameSuffix();
+
+        assertUpdate("CREATE TABLE " + tableName + " (id INT, a ROW(\"UPPER_CASE\" INT, \"lower_case\" INT, \"MiXeD_cAsE\" INT))");
+        assertUpdate("INSERT INTO " + tableName + " VALUES (1, ROW(2, 3, 4)), (5, ROW(6, 7, 8))", 2);
+
+        String expected = "VALUES (2, 3, 4), (6, 7, 8)";
+        assertQuery("SELECT a.UPPER_CASE, a.lower_case, a.MiXeD_cAsE FROM " + tableName, expected);
+        assertQuery("SELECT a.upper_case, a.lower_case, a.mixed_case FROM " + tableName, expected);
+        assertQuery("SELECT a.UPPER_CASE, a.LOWER_CASE, a.MIXED_CASE FROM " + tableName, expected);
+        assertUpdate("DROP TABLE IF EXISTS " + tableName);
+    }
+
+    @Test
+    public void testProjectionPushdownReadsLessData()
+    {
+        // TODO consider moving this in BaseConnectorTest
+        String tableName = "test_projection_pushdown_reads_less_data_" + randomNameSuffix();
+
+        assertUpdate("CREATE TABLE " + tableName + " (id INT, a ROW(b BIGINT, c BIGINT))");
+        assertUpdate("INSERT INTO " + tableName + " SELECT rand(), ROW(rand(), custkey) FROM tpch.sf1.customer", 150000);
```

Comment:
I had an offline conversation with @raunaqmorarka.

---
@krvikash : I can see the change in `EXPLAIN ANALYZ` query. Input size is reduced with dereference pushdwon, however Physical input: 282.59kB is the same (Is it expected?).

@raunaqmorarka : Maybe you're lacking enough unique data to see a difference. With a query like

```
create table ntest(orderkey BIGINT, col ROW(a BIGINT, b BIGINT))  with (format='parquet');
insert into ntest SELECT orderkey, ROW(rand(), rand()) FROM tpch.sf1.orders;
explain analyze select col.a from ntest;
```
I can see clear difference in physical input bytes with and without dereference pushdown in hive connector.

---
To showcase dereference pushdown effectiveness had to use rand() https://github.com/trinodb/trino/pull/17085#discussion_r1182552336. But this test case does not show as effective as it should be, because I am selecting deterministic value field.

`Physical input` is much lower when selecting a random value field than a deterministic value field.

Example:

```
create table test2 (id INT, a ROW(random BIGINT, deterministic BIGINT));
insert into test2 SELECT rand(), ROW(rand(), custkey) FROM tpch.sf1.customer WHERE custkey < 10000;
```
Results with dereference pushdown enabled:

Selecting random value field:
```
trino:dereference_pushdown> explain analyze select a.random from test2;
                                                                                       Query Plan
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Trino version: testversion
 Queued: 627.25us, Analysis: 425.74ms, Planning: 115.76ms, Execution: 493.76ms
 Fragment 1 [SOURCE]
     CPU: 10.75ms, Scheduled: 167.62ms, Blocked 0.00ns (Input: 0.00ns, Output: 0.00ns), Input: 9999 rows (87.88kB); per task: avg.: 9999.00 std.dev.: 0.00, Output: 9999 rows (87.88kB)
     Output layout: [a#random]
     Output partitioning: SINGLE []
     TableScan[table = delta_lake:dereference_pushdown.test2]
         Layout: [a#random:bigint]
         Estimates: {rows: 9999 (87.88kB), cpu: 87.88k, memory: 0B, network: 0B}
         CPU: 11.00ms (100.00%), Scheduled: 167.00ms (100.00%), Blocked: 0.00ns (?%), Output: 9999 rows (87.88kB)
         Input avg.: 9999.00 rows, Input std.dev.: 0.00%
         a#random := a#random:bigint:REGULAR
         Input: 9999 rows (87.88kB), Physical input: 43.47kB, Physical input time: 161830000.00ns
```

Selecting deterministic value field:

```
trino:dereference_pushdown> explain analyze select a.deterministic from test2;
                                                                                      Query Plan
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Trino version: testversion
 Queued: 118.83us, Analysis: 235.45ms, Planning: 100.05ms, Execution: 455.54ms
 Fragment 1 [SOURCE]
     CPU: 6.47ms, Scheduled: 142.90ms, Blocked 0.00ns (Input: 0.00ns, Output: 0.00ns), Input: 9999 rows (87.88kB); per task: avg.: 9999.00 std.dev.: 0.00, Output: 9999 rows (87.88kB)
     Output layout: [a#deterministic]
     Output partitioning: SINGLE []
     TableScan[table = delta_lake:dereference_pushdown.test2]
         Layout: [a#deterministic:bigint]
         Estimates: {rows: 9999 (87.88kB), cpu: 87.88k, memory: 0B, network: 0B}
         CPU: 6.00ms (100.00%), Scheduled: 143.00ms (100.00%), Blocked: 0.00ns (?%), Output: 9999 rows (87.88kB)
         Input avg.: 9999.00 rows, Input std.dev.: 0.00%
         a#deterministic := a#deterministic:bigint:REGULAR
         Input: 9999 rows (87.88kB), Physical input: 81.25kB, Physical input time: 140320000.00ns
```


> integration tests should generally use as small data sets as possible

I will reduce the data sets to 10.

Reply:
> Why rand()?

After discussing with @findinpath, We found that we do not require rand(). Since rand() generates the value 0 or 1, and it seems `Run length encoding (RLE)` of parquet file was playing a huge role in the above example `Selecting random value field:`.

We can avoid the use of `rand()` here. I will update the test.